src/main/java/uk/gov/hmcts/ccd/domain/types/DynamicListValidator.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
+
+@Named
+@Singleton
+public class DynamicListValidator implements BaseTypeValidator {
```

Comment:
I initially tried to reuse the existing validator, but the way the contract as been agreed with I needed a different validator, with fixed list the format comes from definition with dynamic it comes from service in the following format. 
        "dynamicList": {
          "default": {
            "code": "FixedList1",
            "label": "Fixed List 1"
          },
          "dynamic_list_items": [{
            "code": "FixedList1",
            "label": "Fixed List 1"
          }, {
            "code": "FixedList2",
            "label": "Fixed List 2"
          }, {
            "code": "FixedList3",
            "label": "Fixed List 3"
          }, {
            "code": "FixedList4",
            "label": "Fixed List 4"
          }, {
            "code": "FixedList5",
            "label": "Fixed List 5"
          }, {
            "code": "FixedList6",
            "label": "Fixed List 6"
          }, {
            "code": "FixedList7",
            "label": "Fixed List 7"
          }
          ]
        }

Reply:
Because the format is different from normal fixed_list_items I felt it was safer to have a dynamic_list_items plus changing name across of fixed_list_items to list_items which seemed to be like a lot bigger change UI and regression testing across.

Reply:
ok got it 👍